scripts/verify_adv_IM.py

```diff
@@ -32,79 +66,160 @@ def parse_args():
             )
 
-        # check station count, if match, skip rest of checks
-        csv_stations = df.station.unique()
-        # glob for station folders
-        list_folders = [
-            y
-            for y in [os.path.join(im_calc_dir, x) for x in os.listdir(im_calc_dir)]
-            if os.path.isdir(y)
-        ]
-        if len(csv_stations) == len(list_folders):
-            # station folder count matches csv.station.count
-            continue
+        # a quick check to compare station count, will skip all other checks if successful.
+        if simple_check:
+            # using try/except to prevent crash
+            try:
+                df_csv = pd.read_csv(csv_path)
+            except FileNotFoundError:
+                # failed to read a agg csv, leave df_model as is.
+                print("csv for {} does not exist".format(model))
+                continue
+            # check for null/nan
+            if df_csv.isnull().values.any():
+                # agg csv is there, but value has NaN
+                # change all status to unknown
+                df_model.loc["status"] = run_status.unknown.value
+                continue
+
+            # check station count, if match, skip rest of checks
+            csv_stations = df_csv.station.unique()
+            if len(csv_stations) == len(station_list):
+                # station folder count matches csv.station.count
+                print("test")
+                df_model.loc["status"] = run_status.finished.value
+                continue
         # check for logs
-        failed_stations = check_log(list_folders, model, csv_stations, components)
-        if len(failed_stations) != 0:
-            failed.append((model, "Crashed", failed_stations))
-    if len(failed) != 0:
-        print("some runs have failed. models: {}".format(failed))
-        return 1
-    else:
-        print("check passed")
-        return 0
+        check_log(list_folders, model, components, df_model, break_on_fail=True)
+        print(model)
+        df_dict.append((df_model, model))
+
+    #
+    result_code = 0
+    for df, model_name in df_dict:
+        # check if any status >= 3 or != 0
+        if df["status"].ge(run_status.not_finished.value).any():
+            print(f"{model_name} have errors please check the status.csv")
+            result_code += 1
+        if df["status"].eq(run_status.not_started.value).any():
+            print(
+                f"{model_name} has some stations that havent been analysised. please check status.csv"
+            )
+            result_code += 2
+        # sort index by status
+        df.sort_values("status", inplace=True, ascending=False)
+        # map status(int) to string before saving as csv
+        df["status"] = df["status"].map(lambda x: run_status(x).name)
+        status_csv_path = os.path.join(im_calc_dir, "{}_status.csv".format(model_name))
+        df.to_csv(status_csv_path, header=True, index=True)
+    return result_code
```

Comment:
Is this going to be useful once you've gone past one error? It'll give you some idea of the magnitude of the issues, but it'll still be a mix of +=1 and +=2 problems

Reply:
currently it doesnt have any usage. but it can be usefull if we have a external code that calls it and the error code can be easily identified.

Reply:
so by knowing the value, you can tell how many errors were there. ( i.e looking at it in binary, 00, 01, 10, 11)

Reply:
Yeah, but there's a for loop, so you might end up with 10 from 1+1 or from 2, or how ever many entries are in the df_dict